src/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/Manager.cpp

```diff
@@ -106,196 +101,171 @@ CSSISpaceWeather Manager::getCSSISpaceWeatherAt(const Instant& anInstant) const
 
 Array<Integer> Manager::getKp3HourSolarIndicesAt(const Instant& anInstant) const
 {
-    if (!anInstant.isDefined())
-    {
-        throw ostk::core::error::runtime::Undefined("Instant");
-    }
+    std::lock_guard<std::mutex> lock {mutex_};
+    const CSSISpaceWeather* CSSISpaceWeatherPtr = this->accessCSSISpaceWeatherAt(anInstant);
 
-    const Date day = anInstant.getDateTime(Scale::UTC).getDate();
-    const Instant dayInstant = Instant::DateTime(DateTime(day, Time::Midnight()), Scale::UTC);
+    static auto dataFilter = [](const CSSISpaceWeather::Reading& aReading) -> Array<Integer>
```

Comment:
It behaves in a similar way to the Python `filter()` function, i.e. plucking data points from a larger "dict" like structure. Def not exactly the same, but seems like a filtering operation to me?

Reply:
Fair point... IMO filter would (potentially) change the length of the data points.

In either case though, do you think that it would be more readable with an explicit name? That way you don't have to scroll up to read what `dataFilter` is doing.

Reply:
Yeah that's fair. Not opposed to the better readability 👍